src/blockdata/transaction.rs

```diff
@@ -437,11 +438,11 @@ impl Transaction {
     }
 }
 
-impl BitcoinHash for Transaction {
-    fn bitcoin_hash(&self) -> sha256d::Hash {
-        let mut enc = sha256d::Hash::engine();
+impl BitcoinHash<Txid> for Transaction {
+    fn bitcoin_hash(&self) -> Txid {
```

Comment:
Yeah this is not correct. The txid skips the witness data. This should be `Wtxid`. I'm not sure I'm a fan of the `BitcoinHash` trait in this context.